docs/KUBERNETES.md

```diff
@@ -0,0 +1,201 @@
+
+### Installation Prerequisites
+
+* LogDNA Account - Create an account with LogDNA by following our [quick start guide](https://docs.logdna.com/docs/logdna-quick-start-guide).
+* LogDNA Ingestion Key - You can find an ingestion key at the top of [your account's Add a Log Source page](https://app.logdna.com/pages/add-host).
+* Kubernetes cluster running at least version 1.9.
+* Local clone of this repository.
+
+### Installation Steps
+
+1. Navigate to the root directory of the cloned `logdna-agent` repository.
+2. Run the following commands to configure and start the agent:
+```console
+kubectl apply -f k8s/agent-namespace.yaml
+kubectl create secret generic logdna-agent-key -n logdna-agent --from-literal=logdna-agent-key=<YOUR LOGDNA INGESTION KEY>
+kubectl apply -f k8s/agent-resources.yaml
+```
+3. Monitor the pods for startup success:
+```console
+foo@bar:~$ kubectl get pods -n logdna-agent --watch
+NAME                 READY   STATUS    RESTARTS   AGE
+logdna-agent-hcvhn   1/1     Running   0          10s
+```
+
+> :warning: By default the agent will run as root. To run the agent as a non-root user, check out the [run as non-root section](#run-as-non-root).
+
+
+## Upgrading
+
+There are two components that can be upgraded independent of each other for each new version of the agent. While not strictly required, we always recommend upgrading both components together.
+
+### Upgrading the Configuration
+
+Not every version update of the agent makes a change to our supplied configuration yamls. These changes will be outlined in our release page to help you determine if you need to update your configuration.
+
+Due to how the agent has evolved over time, certain versions of the agent configuration yamls require different paths to be updated successfully.
+
+If you are unsure of what version of the configuration you have, you can always check the `app.kubernetes.io/version` label of the DaemonSet:
+
+```console
+foo@bar:~$ kubectl describe daemonset --all-namespaces -l app.kubernetes.io/name=logdna-agent
+Name:           logdna-agent
+Selector:       app=logdna-agent
+Node-Selector:  <none>
+Labels:         app.kubernetes.io/instance=logdna-agent
+                app.kubernetes.io/name=logdna-agent
+                app.kubernetes.io/version=2.2.0
+...
+```
+
+Older versions of our configurations do not provide these labels. In that case, each upgrade path belows provides an example of each configuration to be compared to whats running on your cluster.
+
+#### Upgrading from Configuration v1.x.x or v2.0.x
+
+* **Example Configuration Yamls:**
+  * [v1.x.x](https://raw.githubusercontent.com/logdna/logdna-agent/master/logdna-agent-ds.yaml)
+  * [v2.0.x](https://raw.githubusercontent.com/logdna/logdna-agent/master/logdna-agent-v2.yaml)
+* **Differences:** This configuration does not include the new logdna-agent namespace and is lacking a number of new Kubernetes Objects.
```

Comment:
Maybe "Differences with latest Agent v2"

Reply:
I think the documentation will be easier to maintain and more robust if we don't specify the current version the docs refer to.

Reply:
Fix the "Objects"